test/e2e/kappcontroller/app_status_test.go

```diff
@@ -58,10 +59,17 @@ spec:
 	defer cleanUpApp()
 
 	logger.Section("deploy", func() {
-		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, e2e.RunOpts{StdinReader: strings.NewReader(appYaml), AllowError: true})
+		out, err := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, e2e.RunOpts{StdinReader: strings.NewReader(appYaml), AllowError: true})
+		// it's supposed to error, but it's also supposed to create the three supporting resources successfully:
+		assert.Error(t, err)
+		assert.Contains(t, out, "ok: reconcile role/kappctrl-e2e-ns-role (rbac.authorization.k8s.io/v1) namespace: kappctrl-test")
+		assert.Contains(t, out, "ok: reconcile rolebinding/kappctrl-e2e-ns-role-binding (rbac.authorization.k8s.io/v1) namespace: kappctrl-test")
+		assert.Contains(t, out, "ok: reconcile serviceaccount/kappctrl-e2e-ns-sa (v1) namespace: kappctrl-test")
+
```

Comment:
namespace cant be hard coded since it comes from kapp class. given below assertion im not sure it's worth adding this. alternatively you can may be assert on error message from below inside expectedStatus.

Reply:
i'm just doing substring assertions so it's appealing to amend by shortening the string to exclude the namespace.   I'd still like to capture the information that those three resources should be successfully created either by asserting here or below in the output of the inspect.

Reply:
not pretty looking, but ok.